Crash involving Geofencing in Background

## Steps to Reproduce
1. App is running in the background

## Problem
App crashes

### Environment
* Phone: Pixel
* OS version: Android Oreo 8.1
* TUM Campus App version: current master
* Language: English

### Related Issue
#460 

### Stack trace
```
02-14 08:37:38.809 5934-5934/de.tum.in.tumcampus E/AndroidRuntime: FATAL EXCEPTION: main
                                                                   Process: de.tum.in.tumcampus, PID: 5934
                                                                   java.lang.RuntimeException: Unable to start receiver de.tum.in.tumcampusapp.services.GeofencingStartupReceiver: java.lang.IllegalArgumentException: Scheduled service ComponentInfo{de.tum.in.tumcampus/de.tum.in.tumcampusapp.services.GeofencingRegistrationService} does not require android.permission.BIND_JOB_SERVICE permission
                                                                       at android.app.ActivityThread.handleReceiver(ActivityThread.java:3194)
                                                                       at android.app.ActivityThread.-wrap17(Unknown Source:0)
                                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1672)
                                                                       at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                       at android.os.Looper.loop(Looper.java:164)
                                                                       at android.app.ActivityThread.main(ActivityThread.java:6494)
                                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                                       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
                                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
                                                                    Caused by: java.lang.IllegalArgumentException: Scheduled service ComponentInfo{de.tum.in.tumcampus/de.tum.in.tumcampusapp.services.GeofencingRegistrationService} does not require android.permission.BIND_JOB_SERVICE permission
                                                                       at android.os.Parcel.readException(Parcel.java:2008)
                                                                       at android.os.Parcel.readException(Parcel.java:1950)
                                                                       at android.app.job.IJobScheduler$Stub$Proxy.enqueue(IJobScheduler.java:211)
                                                                       at android.app.JobSchedulerImpl.enqueue(JobSchedulerImpl.java:53)
                                                                       at android.support.v4.app.JobIntentService$JobWorkEnqueuer.enqueueWork(JobIntentService.java:342)
                                                                       at android.support.v4.app.JobIntentService.enqueueWork(JobIntentService.java:522)
                                                                       at android.support.v4.app.JobIntentService.enqueueWork(JobIntentService.java:500)
                                                                       at de.tum.in.tumcampusapp.services.GeofencingRegistrationService$Companion.startGeofencing(GeofencingRegistrationService.kt:64)
                                                                       at de.tum.in.tumcampusapp.services.GeofencingStartupReceiver.onReceive(GeofencingStartupReceiver.kt:21)
                                                                       at android.app.ActivityThread.handleReceiver(ActivityThread.java:3187)
                                                                       at android.app.ActivityThread.-wrap17(Unknown Source:0) 
                                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1672) 
                                                                       at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                       at android.os.Looper.loop(Looper.java:164) 
                                                                       at android.app.ActivityThread.main(ActivityThread.java:6494) 
                                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                                       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
                                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
```

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -664,7 +664,8 @@
             </intent-filter>
         </service>
 
-        <service android:name=".services.GeofencingRegistrationService"/>
+        <service android:name=".services.GeofencingRegistrationService"
+                 android:permission="android.permission.BIND_JOB_SERVICE"/>
 
         <receiver android:name=".services.GeofencingUpdateReceiver"/>
 

